src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1127,78 +1135,19 @@ def update_vm(cmd, resource_group_name, vm_name, os_disk=None, disk_caching=None
     if proximity_placement_group is not None:
         vm.proximity_placement_group = {'id': proximity_placement_group}
 
+    if workspace is not None:
+        workspace_id = _prepare_workspace(cmd, resource_group_name, workspace)
+        workspace_name = parse_resource_id(workspace_id)['name']
+        _set_log_analytics_workspace_extension(cmd=cmd,
+                                               resource_group_name=resource_group_name,
+                                               vm=vm,
+                                               vm_name=vm_name,
+                                               workspace_name=workspace_name)
+        os_type = vm.storage_profile.os_disk.os_type.value if vm.storage_profile.os_disk.os_type else None
```

Comment:
what if set_extension failed here ?

Reply:
If it fail with an error, this command would stop. I think it's ok. If it fail with warning. It's still ok to set data source. I check the logic of `set_extension` and look like it will only fail with an error. Is there any concern?